### PR TITLE
feat(ecs): add port mappings to containers with props

### LINKED ISSUE #13261: (ecs): add a `portMappings` prop to `ContainerDefinition`
I'd like to allow the user to define port mappings when adding containers to a task definition.

### Use Case

This would allow users to add containers in a more declarative fashion.

### Proposed Solution

Add an optional `portMappings` prop to `ContainerDefinitionOptions` and have `ContainerDefinition` add the port mappings at construction time. After, I'd be able to do this:

```ts
taskDefinition.addContainer('cont', {
  image: ecs.ContainerImage.fromRegistry('test'),
  memoryLimitMiB: 1024,
  // New option:
  portMappings: [{ containerPort: 80 }],
});
```

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by misterjoshua**
This PR adds an optional `portMappings` property to `ContainerDefinitionOptions` so that users can add port mappings when they add a container to a `TaskDefinition`.

Closes #13261

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nathanpeck** [comment]
+1 This one looks good to me. I'll get a second opinion from @SoManyHs or @piradeepk just to be on the safe side though

**Comment by piradeepk** [comment]
Looks good to me too. @misterjoshua do you mind updating one of the existing integ tests to define container port mappings in the constructor, rather than after construction?

File: packages/@aws-cdk/aws-ecs/README.md [hunk 1]
```diff
@@ -237,6 +237,14 @@ const container = ec2TaskDefinition.addContainer("WebContainer", {
...
+taskDefinition.addContainer("WebContainer", {
+  image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+  memoryLimitMiB: 1024,
+  portMappings: [{ containerPort: 3000 }]
+})
+```
+
 ```ts
 container.addPortMappings({
   containerPort: 3000
```
Comment (piradeepk) [inline packages/@aws-cdk/aws-ecs/README.md:250]: Can you add a comment above both to explain that there are 2 ways of defining the port mapping?

**Comment by misterjoshua** [comment]
@piradeepk I've made the change you requested.

> Looks good to me too. @misterjoshua do you mind updating one of the existing integ tests to define container port mappings in the constructor, rather than after construction?

@piradeepk Done! :)

File: packages/@aws-cdk/aws-ecs/README.md (hunk 1 above)
Comment (misterjoshua) [reply packages/@aws-cdk/aws-ecs/README.md:250]: Oh sure. Just saw this. Let me get right on that.

**Comment by piradeepk** [comment]
@misterjoshua thanks for the patience. I was wondering if you could add one more unit test case in which you add a port mapping to the constructor and also call addPortMapping with a different port mapping (to verify that both mappings are included, and that one does not overwrite the other).

**Comment by misterjoshua** [comment]
> @misterjoshua thanks for the patience. I was wondering if you could add one more unit test case in which you add a port mapping to the constructor and also call addPortMapping with a different port mapping (to verify that both mappings are included, and that one does not overwrite the other).

@piradeepk This is done. Please let me know if you need any more changes.

**Review by piradeepk — APPROVED** [review]
This is awesome! Thanks for contributing and for addressing all of the feedback @misterjoshua!!! Ship it! 💯

**Pull request merged**